client/blocks/site-preview/index.jsx

```diff
@@ -61,22 +40,33 @@ function SitePreview( { className } ) {
 				'frame-nonce': selectedSiteNonce ?? '',
 				cachebust: previewCount,
 			},
-			basePreviewUrl
+			previewUrl
 		);
 	}
 
 	return (
 		<WebPreview
 			className={ className }
 			previewUrl={ formatPreviewUrl() }
-			externalUrl={ basePreviewUrl }
+			externalUrl={ previewUrl }
 			showExternal={ true }
 			showClose={ true }
 			showPreview={ showPreview }
-			onClose={ () => dispatch( closePreview() ) }
+			onClose={ () => {
+				dispatch( closePreview() );
+				incrementPreviewCount();
+			} }
```

Comment:
Yes, this is a big improvement. But I think we can go further, not necessarily in this PR, but in a follow-up, and move the preview count to Redux.

In the end, there would be just two Redux actions:
- `setPreview( url, siteId )` sets the url and site, and generates a random cache buster. And it would also be a thunk that sets the layout focus and therefore displays the preview.
- `clearPreview()` would unset the url and the site, and close the preview by resetting the layout focus.

Then the `<SitePreview>` component reads the URL, site and cache buster and uses all three to construct the real preview URL, with busters and site nonces.

The point of the `previewCount` cache buster is that if I repeatedly open and close a preview of the same post (e.g., from the Posts page), I want the iframe to load a new document from the server each time (the post could have been updated in the meantime) and avoid any cached version. That's why add the random cache buster param.